R/utils.R

```diff
@@ -292,6 +292,33 @@ filter_null <- function(x) {
   x[!vapply(x, is.null, logical(1))]
 }
 
+# the following function attempts to prune remote CSS
+# or local CSS/JS dependencies that either should not
+# be resolved to local R package paths, or which have
+# insufficient information to do so. 
+#
+# this attempts to avoid cryptic errors produced by
+# get_package_mapping, which requires three parameters:
+# -- the script name (i.e. x$script below)
+# -- the package name from the URL (i.e. x$package)
+# -- the list of dependencies (i.e. deps)
+#
+# within get_package_mapping, x$package is also required,
+# so deps missing it here are assigned NULL and then
+# filtered out by the subsequent vapply statement
+assert_valid_dependencies <- function(deps) {
```

Comment:
`assert` says to me that this function performs a test - which would then either throw an error or return a boolean. So I'd call this something like `clean_dependencies` instead.

Reply:
I agree, but I realized that I should add a check for non-stylesheet deps that ensures the essential details are present, and throws an error if they aren't.

Since I will be adding at least one test, should I still rename this function?

Reply:
It's more about what the function returns than anything else - you can have tests throwing an error inside any function, but if it's *name* is `assert_*` then it should have either no return value or it should return a boolean:
`all_deps_ok = assert_valid_dependencies(deps)`
Above what you effectively have is:
`valid_dep_list = assert_valid_dependencies(deps)`
where the verb `assert` just doesn't match the return value.

Reply:
fixed in f85a8da0abfacabc85c2cfd865dd8d275de8d073